src/plugins/guided_onboarding/public/services/api.ts

```diff
@@ -312,7 +312,7 @@ export class ApiService implements GuidedOnboardingApi {
       const currentGuide: GuideState = {
         guideId,
         isActive: true,
-        status: isLastStep(guideId, stepId) ? 'ready_to_complete' : 'in_progress',
+        status: getGuideStatusOnStepCompletion(guideState, guideId, stepId),
```

Comment:
Thanks a lot for adding the test to the panel component but ideally we'd need a unit test for the api service. Could you please add a test to `api.test.ts` that would check that the guide is updated correctly, when the last step (which is manual completion) is completed.

Reply:
Addressed with https://github.com/elastic/kibana/pull/144221/commits/0485eb16318d6b7b3a8c841b97a6c7de4f03223a. Thanks for pointing this out @yuliacech!